### PR TITLE
Migrate from dicts to classes

### LINKED ISSUE #8: Migrate from dicts to classes
Considering that Hawkular-Metrics API will live and change, handling different dicts is becoming an issue and support in the IDEs is better for classes and objects.

### PR DESCRIPTION
**Pull request by rubenvp8510**
Fixes #8, I'm not sure if this is will be done or not, but I did it, my main concern here is backward compatibility.

### CODE REVIEW

File: hawkular/metrics.py [hunk 1]
```diff
@@ -123,19 +171,23 @@ def put(self, data):
...
         for l in r:
-            self._post(self._get_metrics_raw_url(self._get_url(l)), r[l],parse_json=False)
+            serialized_json = self._serialize_object(r[l])
+            self._post(self._get_metrics_raw_url(self._get_url(l)), serialized_json, parse_json=False)
 
     def push(self, metric_type, metric_id, value, timestamp=None):
         """
         Pushes a single metric_id, datapoint combination to the server.
-
+http://forums.thefashionspot.com/f96/tall-models-30100-26.htmladriana lima muy alta
```
Comment (burmanm) [inline hawkular/metrics.py]: ;)

**Comment by burmanm** [comment]
I'm not sure why all the old methods should be deleted, they could remain there still (for backwards compatibility) and only map to the newer methods. At the same time we could fix the newer APIs to allow for example easier use of tagNames with dot in them.

**Comment by rubenvp8510** [comment]
Ok, that could work

**Comment by rubenvp8510** [comment]
I added a flag for switch between objects and dictionaries because most of the changes were in return values (return objects instead of dictionaries), Also added tests for it...

WDYT?

**Comment by rubenvp8510** [comment]
Also, I'm not sure if this solves #42  too.

**Comment by rubenvp8510** [comment]
Also, I'm not sure if this solves #42  too.

**Pull request closed without merging**